### DISCUSSION TITLE
No BI Tools work with PlanetScale

### DISCUSSION THREAD

**Discussion by shelbyt** [planetscale/discussion#186] [Q&A] [answered]
I cannot use any major BI tool, powerBi, Metabase, Tableau, Google Data Studio to connect with the PlanetScale database. I tried all of these tools using the MySQL connector but get failure on everyone with/out SSL. I can't believe this isn't being talked about more as a major limitation. I haven't seen any discussions around this.

How are folks visualizing their data today?

**Comment by orware** [comment]
Hello @shelbyt!

We definitely have users that have been testing things out with PlanetScale and Metabase in recent months (there's a small internal adjustment we need to make to switch the database over to our new Gen4 Query Planner, but those users didn't have trouble connecting to the PlanetScale database itself as far as I can recall, but the Gen4 Query Planner adjustment was needed for the additional operations that Metabase wanted to use after adding in a new database...for example, Metabase runs some metadata collection queries to learn more about the tables/fields in the database you are adding in).

As far as I can recall, we haven't had any users report trying to use PowerBI, Tableau, or Google Data Studio so far.

The SSL connection to the PlanetScale database should work however (so long as each of those tools supports the option). I'd be curious if any of those tools might require the Gen4 Query Planner feature to be enabled as well after confirming that an SSL connection is able to be established successfully.

If you can please reach out to us via:
https://planetscale.com/support

So we can assist you further (you can't directly attach anything upon initial submission of your ticket on that page, but once the ticket is created you can reply with additional screenshots and other info and we can see if we can get some of those BI tools working for you with your PlanetScale database).

**Reply by ghost** [reply]
I need to create a connection with a user who has only select privilege, any tips?

This doesn't work:
```sql
CREATE USER 'user_test'@'%' IDENTIFIED BY 'user_test_pass';
GRANT SELECT, SHOW VIEW ON `my-table` . * TO 'user_test'@'%';
```

**Reply by kiwioverseas** [reply]
Did you find a way to create select only users? I'm having the same issue (1105 error) and found this thread.

**Comment by nirnejak** [comment]
We recently added full support for PlanetScale on Draxlr(https://draxlr.com), Both for the Query Builder and the Raw SQL Queries.

**Reply by nirnejak** [reply]
we also get around some of the limitations which usually break some of the BI tools like the option to add custom join in the query builder(since you can't add Foreign Key Constraints, bi tools won't fetch the relation between tables)

**Comment by dluan** [comment]
I also tried using Google Data Studio, and it doesn't connect at all.

Has anyone done it, or is it planned to make this possible?

**Reply by fooforge** [reply]
Hey @dluan, I saw that you also opened a ticket. I have responded to it earlier today, but am also going to drop a copy of my response here. Hopefully it's going to be of use for others as well.

Updated instructions:
https://github.com/planetscale/discussion/discussions/186#discussioncomment-5583073

Note that if we ever change our SSL certificate provider you will need to update this certificate. We don't have any plans to change our provider Let's Encrypt anytime soon, but I wanted to mention it nonetheless.

Yet another way to connect PlanetScale with external resources is Airbyte, an Open Source data integration service. We currently only support [the self-hosted version](https://docs.airbyte.com/quickstart/deploy-airbyte/), but setting it up is relatively trivial. We have that process [documented here](https://docs.planetscale.com/integrations/airbyte).

**Reply by dluan** [reply]
Thanks, @fooforge, it worked perfectly with the certificate you provided.

I'll reread the documentation on that part again, because I must have done something wrong and given the wrong certificate when trying to connect to Data Studio.

**Reply by eelco** [reply]
Is this still supported? I tried it just now, but I’m getting errors at the Looker Studio side unfortunately.

**Reply by fooforge** [reply]
@eelco I have just verified that this still works as expected. It would have been a surprise if it didn't.

All you will need to do is to upload our CA's root certificate. I am going to attach it as a `.txt` file on my reply. You will need to rename the file to `.pem`, though, as Data Studio will not allow you to upload it otherwise.

![CleanShot 2023-04-11 at 18 15 41](https://user-images.githubusercontent.com/959793/231225246-60a04427-24a5-4964-acc2-e2755b8e3ad1.png)

[isrgx1.txt](https://github.com/planetscale/discussion/files/11202687/isrgx1.txt)

**Comment by fooforge** [comment, marked as answer]
To answer the original question, we have [shipped PlanetScale Connect](https://docs.planetscale.com/integrations/airbyte) at the end of May which should help with extracting data from PlanetScale and pushing it to other destinations for analysis, transformation, archival, etc.

The only caveat is that you cannot configure PlanetScale as a destination in Airbyte yet, only as a source. And, although a MySQL connector exists, it requires the use of `LOAD DATA LOCAL INFILE` and Common Table Expressions, neither of which are currently supported in Vitess or PlanetScale so, sadly, that's not an option either.

I also want to highlight @nirnejak's comment [in this thread](https://github.com/planetscale/discussion/discussions/186#discussioncomment-3120260). Thanks for adding support for PlanetScale! :-)

Last but not least, if you're running into connectivity issues, reach out to us at https://support.planetscale.com/ and we will see what we can do to help.

**Reply by kiwioverseas** [reply]
Thanks for the update @fooforge. WIll PlanetScale will be available as a source in Airbyte Cloud when it's out of beta? The page you linked to notes it's only available in Airbyte Open Source.

**Reply by fooforge** [reply]
@kiwioverseas We would love to, but it's not so easy.

Airbyte is [used to take over maintenance](https://github.com/airbytehq/airbyte/tree/master/docs/connector-development) for connectors once they have been written and submitted to their monorepo. However, we did not just want to throw our [Open Source connector](https://github.com/planetscale/airbyte-source) _over the wall_ and be done with it, but rather maintain it ourselves -- especially as PlanetScale is still rapidly expanding its feature set and functionality. It just didn't feel right to us at this time.

My latest information is that Airbyte is working out the operational and legal bits, though, to allow other companies to submit their connectors to Airbyte Cloud and to maintain them themselves. Once that is in place we're definitely going to revisit.

**Reply by paulknisely** [reply]
@fooforge Any update regarding this matter?

**Reply by snario** [reply]
Checking in on Airbyte Cloud support!

**Reply by jgreet** [reply]
The PlanetScale connector is still only available in self-hosted Airbyte. It looks like there was some some movement toward including it last year but that seems to have fizzled out https://github.com/airbytehq/airbyte/pull/24965.

If you need a hosted solution perhaps one of the other ETL platforms supported by PlanetScale Connect will do? 

https://planetscale.com/docs/concepts/planetscale-connect

**Comment by murphyslaw2781** [comment]
Could I get help with Databox? It requires more than the SSL CA

![image](https://github.com/planetscale/discussion/assets/14646107/9df0cc1c-7b8c-48b5-b3f5-a2addceed1ef)

**Reply by murphyslaw2781** [reply]
@fooforge

**Reply by patrickclover** [reply]
Has anyone found a solution to this?

**Comment by nikita-doronin** [comment]
Example how to connect PlanetScale to Google Looker Studio - works for me:

1) Select MySQL connector at Google Looker.

2) Put you credentials as:
Host name or IP: aws.connect.psdb.cloud
Database: your-database-name
Username: your-username
Password: your-password
Port could be empty.

3) Tick "Enable SSL"

4) Download the **Active ISRG Root X1 "pem"** certificate from [HERE](https://letsencrypt.org/certs/isrgrootx1.pem) and insert this file in attachment as server certificate.

Enjoy your Google Looker real-time dashboard!
<img width="420" alt="SSL" src="https://github.com/planetscale/discussion/assets/108275340/5be77fe0-cf74-43b1-8425-878998337e37">

**Comment by dragonsahead** [comment]
Adding here a comment about Metabase: https://github.com/planetscale/discussion/discussions/161#discussioncomment-12488925. We really need help from the Planetscale team
